// @emailjs/nodejs/cjs/methods/send/send.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.send = void 0;
const store_1 = require("../../store/store");
const validate_params_1 = require("../../utils/validate_params");
const send_json_1 = require("../../api/send_json");
/**
 * Send a template to the specific EmailJS service
 * @param {string} serviceID - the EmailJS service ID
 * @param {string} templateID - the EmailJS template ID
 * @param {object} templatePrams - the template params, what will be set to the EmailJS template
 * @param {string} publicKey - the EmailJS public key
 * @returns {Promise<string>}
 */
const send = (serviceID, templateID, templatePrams, publicKey) => {
    const pubKey = publicKey || store_1.store._userID;
    (0, validate_params_1.validateParams)(pubKey, serviceID, templateID);
    const params = {
        lib_version: '1.0.0',
        user_id: pubKey,
        service_id: serviceID,
        template_id: templateID,
        template_params: templatePrams,
    };
    return (0, send_json_1.sendJSON)(JSON.stringify(params));
};
exports.send = send;
